src/js/widgets/recommended.ts

```diff
@@ -19,9 +46,9 @@ function constructUrl( apiUrl, permalink, personalized ) {
 	return `${ apiUrl }&url=${ encodeURIComponent( permalink ) }`;
 }
 
-function constructWidget( widget ) {
-	const apiUrl = widget.getAttribute( 'data-parsely-widget-api-url' );
-	const permalink = widget.getAttribute( 'data-parsely-widget-permalink' );
+function constructWidget( widget: Element ): WidgetOptions {
+	const apiUrl = widget.getAttribute( 'data-parsely-widget-api-url' ) as string;
+	const permalink = widget.getAttribute( 'data-parsely-widget-permalink' ) as string;
```

Comment:
The `as string` is dangerous here because it suppresses the `null` case. We should save `as` for last-resort cases where we're certain it will be the type we inject. Better to acknowledge that it could be null and set a default value. Or maybe it should throw an error.
```suggestion
	const apiUrl = widget.getAttribute( 'data-parsely-widget-api-url' ) || '';
	const permalink = widget.getAttribute( 'data-parsely-widget-permalink' ) || '';
```